### PR TITLE
[llvm][NFC] Autoupdater x86 detection

### PR DESCRIPTION
**Pull request by urnathan**
The x86 intrinsic detector has grown over time by (mostly) appending new entries. However, sorting the entries in colation order makes it easier to see which intrinsics are affected -- and as each has a note about when it was added, no information is lost. But more than that, it permits common prefix consumption, reducing the number of tests that are performed. We replace 324 string compares with 14 prefix tests (and subsequent compares if one of those hits).

### CODE REVIEW

**Review by nikic — APPROVED** [review]
LGTM, though I didn't try to verify that nothing got lost in the transition.

File: llvm/lib/IR/AutoUpgrade.cpp [hunk 1]
```diff
@@ -122,333 +122,363 @@ static bool ShouldUpgradeX86Intrinsic(Function *F, StringRef Name) {
...
+          Name == "addcarry.u64" ||         // Added in 8.0
+          Name == "addcarryx.u32" ||        // Added in 8.0
+          Name == "addcarryx.u64" ||        // Added in 8.0
+          Name == "subborrow.u32" ||        // Added in 8.0
+          Name == "subborrow.u64" ||        // Added in 8.0
+          Name.starts_with("vcvtph2ps.") || // Added in 11.0
+          Name == "xop.vpcmov" ||           // Added in 3.8
+          Name == "xop.vpcmov.256" ||       // Added in 5.0
+          Name.starts_with("xop.vpcom") ||  // Added in 3.2, Updated in 9.0
+          Name.starts_with("xop.vprot"));   // Added in 8.0
```
Comment (RKSimon) [inline llvm/lib/IR/AutoUpgrade.cpp:481]: Not worth pulling out the "xop." prefix?

**Review by RKSimon — APPROVED** [review]
LGTM with one minor query

File: llvm/lib/IR/AutoUpgrade.cpp (hunk 1 above)
Comment (urnathan) [reply llvm/lib/IR/AutoUpgrade.cpp]: yeah, I was equivocating on that one.  Here's an update that pulls it out.

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Pull request merged**